### ISSUE TITLE
Compile on macOS failed.

### ISSUE THREAD

**Issue by liketemple** [cocos2d/cocos2d-x#16589]
- cocos2d-x version: 3.13.1
- devices test on: macOS 10.12, macbook pro 
- developing environments
  - NDK version:
  - Xcode version: 8.0
  - VS version:
  - brower type and version:

Steps to Reproduce:
1. Download the 3.13.1 zip from github
2. #./download-deps.sh
3. #./setup.sh
4. #./source ~/.bash_profile
5. #cmake .
6. #make -j8

Then get error like:
[100%] Linking CXX executable ../../bin/cpp-tests.app/Contents/MacOS/cpp-tests
Undefined symbols for architecture x86_64:
  "FontUtils::_calculateTextAlignment(cocos2d::Device::TextAlign)", referenced from:
      cocos2d::Device::getTextureDataForText(char const_, cocos2d::FontDefinition const&, cocos2d::Device::TextAlign, int&, int&, bool&) in libcocos2dInternal.a(CCDevice-mac.mm.o)
  "FontUtils::_calculateParagraphStyle(bool, int)", referenced from:
      cocos2d::Device::getTextureDataForText(char const_, cocos2d::FontDefinition const&, cocos2d::Device::TextAlign, int&, int&, bool&) in libcocos2dInternal.a(CCDevice-mac.mm.o)
  "FontUtils::_calculateTextDrawStartWidth(cocos2d::Device::TextAlign, CGSize, CGSize)", referenced from:
      cocos2d::Device::getTextureDataForText(char const_, cocos2d::FontDefinition const&, cocos2d::Device::TextAlign, int&, int&, bool&) in libcocos2dInternal.a(CCDevice-mac.mm.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *_\* [bin/cpp-tests.app/Contents/MacOS/cpp-tests] Error 1
make[1]: **\* [tests/cpp-tests/CMakeFiles/cpp-tests.dir/all] Error 2
make: **\* [all] Error 2

Any help would be grateful! thx!

**Comment by slackmoehrle** [comment]
I am working on this testing, see my running thread here: http://discuss.cocos2d-x.org/t/xcode-8-and-ios-10-testing/32164 I will be updating it more today as I continue testing.

**Comment by slackmoehrle** [comment]
I see the same thing in my testing: 

``` sh
[100%] Linking CXX executable ../../bin/cpp-tests.app/Contents/MacOS/cpp-tests
Undefined symbols for architecture x86_64:
  "FontUtils::_calculateTextAlignment(cocos2d::Device::TextAlign)", referenced from:
      cocos2d::Device::getTextureDataForText(char const*, cocos2d::FontDefinition const&, cocos2d::Device::TextAlign, int&, int&, bool&) in libcocos2dInternal.a(CCDevice-mac.mm.o)
  "FontUtils::_calculateParagraphStyle(bool, int)", referenced from:
      cocos2d::Device::getTextureDataForText(char const*, cocos2d::FontDefinition const&, cocos2d::Device::TextAlign, int&, int&, bool&) in libcocos2dInternal.a(CCDevice-mac.mm.o)
  "FontUtils::_calculateTextDrawStartWidth(cocos2d::Device::TextAlign, CGSize, CGSize)", referenced from:
      cocos2d::Device::getTextureDataForText(char const*, cocos2d::FontDefinition const&, cocos2d::Device::TextAlign, int&, int&, bool&) in libcocos2dInternal.a(CCDevice-mac.mm.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/cpp-tests.app/Contents/MacOS/cpp-tests] Error 1
make[1]: *** [tests/cpp-tests/CMakeFiles/cpp-tests.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by zzzjim** [comment]
Build with no tests no javascript no lua

cmake -G "Unix Makefiles" -DBUILD_CPP_EMPTY_TEST=OFF -DBUILD_CPP_TESTS=OFF -DBUILD_LUA_LIBS=OFF -DBUILD_JS_LIBS=OFF ..

**Comment by minggo** [comment]
@liketemple why not use Xcode to build for Mac?

**Comment by leon8zhang** [comment]
I like idea. I like clion.

**Comment by MrCapone** [comment]
Fix for this issue #16636

**Comment by ricardoquesada** [comment]
f
